src/Post_Meta/Meta_Map.php

```diff
@@ -19,13 +19,14 @@ class Meta_Map {
 	 *
 	 * @phpstan-ignore-next-line
 	 */
-	private $post_type = '';
+	private $post_type;
 
 	/** @var Meta_Group[] */
-	private $keys = [];
+	private $keys;
 
 	public function __construct( $post_type ) {
 		$this->post_type = $post_type;
+		$this->keys = [];
```

Comment:
Prefer to have the default defined in the property as it's evaluated at compile time vs this being evaluated at run time, taking advantage of the opcaching.

Reply:
Okay-okay. But it is not possible to read a property without initiation :)

Reply:
That's correct, at which point that property will have long been in the opcache already.

Reply:
The instructor in the open-source school said to assign values to properties in the constructor.

Reply:
When those properties are coming in through via constructor arguments, absolutely. The keys property isn't being passed in as a dependency.